server/runtime/src/main/java/org/infinispan/server/configuration/security/CredentialStoresConfigurationBuilder.java

```diff
@@ -0,0 +1,66 @@
+package org.infinispan.server.configuration.security;
+
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+import org.infinispan.commons.configuration.Builder;
+import org.infinispan.server.Server;
+import org.wildfly.security.credential.Credential;
+
+/**
+ * @author Tristan Tarrant &lt;tristan@infinispan.org&gt;
+ * @since 12.0
+ **/
+public class CredentialStoresConfigurationBuilder implements Builder<CredentialStoresConfiguration> {
+
+   private final SecurityConfigurationBuilder securityConfigurationBuilder;
+   private Map<String, CredentialStoreConfigurationBuilder> credentialStores = new LinkedHashMap<>(2);
+
+   public CredentialStoresConfigurationBuilder(SecurityConfigurationBuilder securityConfigurationBuilder) {
+      this.securityConfigurationBuilder = securityConfigurationBuilder;
+   }
+
+   public CredentialStoreConfigurationBuilder addCredentialStore(String name) {
+      CredentialStoreConfigurationBuilder credentialStoreBuilder = new CredentialStoreConfigurationBuilder(this);
+      credentialStores.put(name, credentialStoreBuilder);
+      return credentialStoreBuilder;
+   }
+
+   public <C extends Credential> C getCredential(String store, String alias, Class<C> type) {
+      CredentialStoreConfigurationBuilder credentialStoreConfigurationBuilder;
+      if (store == null) {
+         if (credentialStores.size() == 1) {
```

Comment:
like the comment for `alias`, I would fail if `store` isn't set.

Reply:
Similarly to my answer above: if there is only one credential store, it is unambiguous.